hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -103,6 +104,15 @@ case class HoodieFileIndex(spark: SparkSession,
     endCompletionTime = options.get(DataSourceReadOptions.END_COMMIT.key)) with FileIndex {
 
   @transient protected var hasPushedDownPartitionPredicates: Boolean = false
+  private val isPartitionSimpleBucketIndex = PartitionBucketIndexUtils.isPartitionSimpleBucketIndex(spark.sparkContext.hadoopConfiguration,
+    metaClient.getBasePath.toString)
+
+  @transient private lazy val bucketIndexSupport = if (isPartitionSimpleBucketIndex) {
+    val specifiedQueryInstant = options.get(DataSourceReadOptions.TIME_TRAVEL_AS_OF_INSTANT.key).map(HoodieSqlCommonUtils.formatQueryInstant)
+    new PartitionBucketIndexSupport(spark, metadataConfig, metaClient, specifiedQueryInstant)
```

Comment:
always query from the latest hash config because there is no SI for reader/writers.

Reply:
Same as `TimeTravel` scenarios mentioned above.